adr/0006-adopt-three-tier-documentation-structure.md

```diff
@@ -0,0 +1,67 @@
+# 6. Adopt Three-Tier Documentation Structure
```

Comment:
We should adapt the [`adr/README`](https://github.com/mozilla-mobile/firefox-ios/blob/main/adr/README.md) as well